File: Algorithms/C++/00441-Arranging Coins.cpp
class Solution {
public:
    int arrangeCoins(int n) {
        int coins{1};
        int row{0};
        while(n > 0){
            row++;
            n -= coins;
            coins++;
        }
        return n == 0 ? row : row-1; 
    }
};

/*
    binary search on right subarray
    [1,2,3,4,5]

    5/2
    2/2
    1

    tc: O(n)
    sc: O(1)

    total = 6
    curr = 1
    row = 0

    row 1: 5 - 1 = 4
    row 2: 4 - 2 = 2
    row 3: 2 - 3 = 0 -> return 3

    return 2 
*/

File: Algorithms/Python3/00441-Arranging Coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        row = 0
        coinsUsed = 0
        while coinsUsed <= n:
            row += 1
            coinsUsed += row

        return row - 1

'''
tc: O(logn)
sc: O(1)
binary search approach
(not implemented)

tc: O(n)
sc: O(1)
5
1: 1
2: 1 + 2 = 3
3: 1 + 2 + 3 = 6
return (row) 2

while totalCoinsUsed <= n
    add new rows
    check if amount of coins used > n
return max number of rows created
'''